core/jvm-native/src/main/scala/cats/effect/unsafe/PollingSystem.scala

```diff
@@ -17,6 +17,23 @@
 package cats.effect
 package unsafe
 
+/**
+ * Encapsulates the methods required for managing and interacting with a polling system. Polling
+ * systems are typically used in scenarios such as handling asynchronous I/O or other
+ * event-driven systems, where one needs to repeatedly check (or "poll") some condition or
+ * state.
+ *
+ * This class abstracts the general components and actions of a polling system, such as:
+ *   - The user-facing interface (API) which interacts with the outside world
+ *   - The thread-local data structure used for polling, which keeps track of the internal state
+ *     of the system and its events
+ *   - The lifecycle management methods, such as creating and closing the polling system and its
+ *     components
+ *   - The interaction methods, such as polling events and interrupting the process
```

Comment:
These are the methods that the runtime (e.g. the work-stealing threadpool or single-thread event loop) uses to interact with the polling system. But they are not intended for user interaction.

```suggestion
 *   - The runtime interaction methods, such as polling events and interrupting the process
```